rocketpy/motors/Motor.py

```diff
@@ -145,88 +160,196 @@ def __init__(
+        endBurnData = [(self.burn_time[1], self.thrust(self.burn_time[1]))]
+        clipped_source = np.append(clipped_source, endBurnData, 0)
+        startBurnData = [(self.burn_time[0], self.thrust(self.burn_time[0]))]
+        clipped_source = np.insert(clipped_source, 0, startBurnData, 0)
+
+        return Function(
+            clipped_source,
+            "Time (s)",
+            "Thrust (N)",
+            self.interpolate,
+            "zero",
+        )
 
     def reshapeThrustCurve(
-        self, burnTime, totalImpulse, oldTotalImpulse=None, startAtZero=True
+        self, newBurnTime, totalImpulse, oldTotalImpulse=None, startAtZero=True
     ):
-        """Transforms the thrust curve supplied by changing its total burn time
-        and/or its total impulse, without altering the general shape of the
-        curve. May translate the curve so that thrust starts at time equals 0,
-        without any delays.
+        """Transforms the thrust curve supplied by changing its total
+        burn time and/or its total impulse, without altering the
+        general shape of the curve. May translate the curve so that
+        thrust starts at time equals 0, without any delays.
 
         Parameters
         ----------
-        burnTime : float
-            New desired burn out time in seconds.
+        newBurnTime : float, tuple of float
+            New desired burn time in seconds.
         totalImpulse : float
             New desired total impulse.
         oldTotalImpulse : float, optional
-            Specify the total impulse of the given thrust curve, overriding the
-            value calculated by numerical integration. If left as None, the
-            value calculated by numerical integration will be used in order to
-            reshape the curve.
+            Specify the total impulse of the given thrust curve,
+            overriding the value calculated by numerical integration.
+            If left as None, the value calculated by numerical
+            integration will be used in order to reshape the curve.
         startAtZero: bool, optional
-            If True, trims the initial thrust curve points which are 0 Newtons,
-            translating the thrust curve so that thrust starts at time equals 0.
-            If False, no translation is applied.
+            If True, trims the initial thrust curve points which
+            are 0 Newtons, translating the thrust curve so that
+            thrust starts at time equals 0. If False, no translation
+            is applied.
 
         Returns
         -------
         None
         """
         # Retrieve current thrust curve data points
-        timeArray = self.thrust.source[:, 0]
-        thrustArray = self.thrust.source[:, 1]
+        timeArray, thrustArray = self.thrust.xArray, self.thrust.yArray
+
         # Move start to time = 0
         if startAtZero and timeArray[0] != 0:
             timeArray = timeArray - timeArray[0]
```

Comment:
The behavior described by the docstring suggests that if `startAtZero` is `True`, than we guarantee that `thrust(t) > 0` for `t > 0`. However, this is not enforced by the action performed by the following two lines:

```python
if startAtZero and timeArray[0] != 0:
     timeArray = timeArray - timeArray[0]
```

Option A: enforce documented behavior.

```python
if startAtZero:
    # Get index of first non-zero thrust value
    nonZeroIndex = thrustArray.nonzero()[0][0]
    # Clip timeArray and thrustArray
    nonZeroIndex = max(1, nonZeroIndex) # Avoids problems when subtracting 1 in the next lines
    thrustArray = thrustArray[nonZeroIndex-1, :]
    timeArray = timeArray[nonZeroIndex-1, :]
    timeArray = timeArray  - timeArray[0]
```

Option B: change docstring to reflect current behavior.

```
        startAtZero: bool, optional
            If True, trims the initial thrust curve points which
            are not given, translating the thrust curve so that
            thrust data starts at time equals 0. If False, no translation
            is applied.
```

I have no preference in terms of usefulness of each option. Rationally speaking, option B is simpler and safer, as it is just a subset of option A.